backend/src/transactions/user.go

```diff
@@ -16,3 +15,17 @@ func GetAllUsers(db *gorm.DB) ([]models.User, error) {
 
 	return users, nil
 }
+
+func UpdateUser(db *gorm.DB, id string, user models.User) (models.User, error) {
```

Comment:
```suggestion
func UpdateUser(db *gorm.DB, id string, user models.User) (*models.User, error) {
```

to allow you to return user as nil instead of empty struct